truss-chains/truss_chains/remote.py

```diff
@@ -437,7 +359,64 @@ def push(
     non_entrypoint_root_dir: Optional[str] = None,
     gen_root: pathlib.Path = pathlib.Path(tempfile.gettempdir()),
 ) -> Optional[ChainService]:
-    return _Pusher(options, gen_root).push(entrypoint, non_entrypoint_root_dir)
+    chainlet_artifacts = _Generator(options, gen_root).generate_chainlet_artifacts(
+        entrypoint,
+        non_entrypoint_root_dir,
+    )
+
+    if options.only_generate_trusses:
+        return None
+
+    if isinstance(options, definitions.PushOptionsBaseten):
+        _create_chains_secret_if_missing(options.remote_provider)
+
+        return _create_baseten_chain(options, chainlet_artifacts)
```

Comment:
Nit: I think having the empty line after the return separates this if block better from the elif block (or no empty line at all).